File: PY/prime.py
import time
import math

# Function to check if a number is prime
def is_prime(n: int) -> bool:
    if n < 2:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

# Function to compute the sum of primes up to a given limit
def sum_of_primes(limit: int) -> int:
    return sum(n for n in range(2, limit + 1) if is_prime(n))

limit = 10_000_000  # Adjust for difficulty
start_time = time.time()

sum_primes = sum_of_primes(limit)

duration = time.time() - start_time
print(f"Sum of primes up to {limit}: {sum_primes}")
print(f"Time taken: {duration:.3f} seconds")

File: Java/prime.java
public class prime {
    // Function to check if a number is prime
    public static boolean isPrime(int n) {
        if (n < 2) return false;
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) return false;
        }
        return true;
    }

    // Function to compute the sum of primes up to a given limit
    public static long sumOfPrimes(int limit) {
        long sum = 0;
        for (int i = 2; i <= limit; i++) {
            if (isPrime(i)) {
                sum += i;
            }
        }
        return sum;
    }

    public static void main(String[] args) {
        int limit = 10000000; // Adjust for difficulty
        long startTime = System.nanoTime();

        long sum = sumOfPrimes(limit);

        long duration = (System.nanoTime() - startTime) / 1_000_000;
        System.out.println("Sum of primes up to " + limit + ": " + sum);
        System.out.println("Time taken: " + duration + " ms");
    }
}
